shaders/EditorHexGrid.shader

```diff
@@ -0,0 +1,62 @@
+shader_type spatial;
+render_mode unshaded;
+
+// If false the shader will use generated hex patterns for the texture which
+// generaly looks better and uniform but probably slightly slower
+uniform bool useTexture;
+
+uniform sampler2D gridTexture;
+uniform sampler2D maskTexture;
+uniform vec4 color : hint_color;
+
+uniform vec2 gridSize;
+
+varying vec3 worldPos;
+
+const vec2 s = vec2(1.7320508, 1);
+
+// Hexagonal grid pattern code from https://www.shadertoy.com/view/wtdSzX
```

Comment:
I can't find a license on that code. Unless the code is liberally licensed we can't use it. This entire algorithm needs to be rewritten (the same approach can't be used, but code without a license can't be copied)

Reply:
Whoops, typoed saying that "approach can't be used" when it should have been "approach can be used".

Reply:
Found a shader with a creative commons license and similar approach to that one https://www.shadertoy.com/view/3sSGWt, I think we can use this.

Reply:
Yeah, I'm very noob with shaders currently. Will probably add a todo comment in the file for implementing it whenever I or someone else got around doing that.